src/main/java/org/snomed/cdsservice/model/CDSTrigger.java

```diff
@@ -12,28 +13,35 @@ public class CDSTrigger {
 
 	private final String medicationLabel;
 	private final Collection<Coding> medicationCodings;
-	private final String conditionLabel;
-	private final Collection<Coding> conditionCodings;
+	private final String conditionOrMedicationLabel;
+	private final Collection<Coding> conditionOrMedicationCodings;
 	private final CDSCard card;
 
-	public CDSTrigger(String medicationLabel, Collection<Coding> medicationCodings, String conditionLabel, Collection<Coding> conditionCodings, CDSCard card) {
```

Comment:
I think it would be cleaner to refactor this to separate the cases. CDSTrigger could become an interface or abstract class if needed but a class for each case would be good: MedicationConditionCDSTrigger and MedicationInterationCDSTrigger

Reply:
used abstract class and created two classes namely MedicationConditionCDSTrigger and MedicationInterationCDSTrigger